golang/pkg/crane/crux/deploy.go

```diff
@@ -31,7 +33,19 @@ func GetDeployments(ctx context.Context, namespace string) []*common.ContainerSt
 		log.Error().Err(err).Stack().Send()
 	}
 
-	return mapper.MapKubeDeploymentListToCruxStateItems(list, svc)
+	podsByDeployment := make(map[string][]corev1.Pod)
```

Comment:
It would be a bad decision to outsource this to a new function? (and also test it with unit tests)

Reply:
This is the only place where it is currently used, but if you think it'd be a good idea to move it to a separate function I could do it.